master/internal/resourcemanagers/fitting_methods_test.go

```diff
@@ -8,28 +8,94 @@ import (
 	"github.com/determined-ai/determined/master/pkg/actor"
 )
 
-func consumeSlots(agent *agentState, consume int) *agentState {
-	req := &AllocateRequest{
-		SlotsNeeded:    consume,
-		NonPreemptible: false,
-	}
-	container := newContainer(req, agent, req.SlotsNeeded)
-	agent.allocateFreeDevices(req.SlotsNeeded, container.id)
-	return agent
+func newIntPtr(n int) *int {
+	return &n
 }
 
 func TestBestFit(t *testing.T) {
 	system := actor.NewSystem(t.Name())
-	assert.Equal(t, BestFit(nil, consumeSlots(newMockAgent(t, system, "agent1", 1, ""), 1)), 1.0)
-	assert.Equal(t, BestFit(nil, consumeSlots(newMockAgent(t, system, "agent2", 1, ""), 0)), 0.5)
-	assert.Equal(t, BestFit(nil, consumeSlots(newMockAgent(t, system, "agent3", 9, ""), 0)), 0.1)
-	assert.Equal(t, BestFit(nil, consumeSlots(newMockAgent(t, system, "agent4", 10, ""), 1)), 0.1)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 0},
+		newFakeAgentState(t, system, "agent1", "", 0, 0, nil, 0),
+	), 0.0)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 0},
+		newFakeAgentState(t, system, "agent2", "", 0, 0, nil, 1),
+	), 1.0)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 0},
+		newFakeAgentState(t, system, "agent3", "", 0, 0, newIntPtr(2), 0),
+	), 1.0/3.0)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 0},
+		newFakeAgentState(t, system, "agent4", "", 0, 0, newIntPtr(2), 1),
+	), 0.5)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 1},
+		newFakeAgentState(t, system, "agent5", "", 1, 1, nil, 0),
+	), 1.0)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 1},
+		newFakeAgentState(t, system, "agent6", "", 1, 0, nil, 0),
+	), 0.5)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 1},
+		newFakeAgentState(t, system, "agent7", "", 9, 0, nil, 0),
+	), 0.1)
+	assert.Equal(t, BestFit(
+		&AllocateRequest{SlotsNeeded: 1},
+		newFakeAgentState(t, system, "agent8", "", 10, 1, nil, 0),
+	), 0.1)
```

Comment:
(Fixing this is beyond the scope of this PR). 

The fact that this test and the one above have the same affinity score seems wrong to me (shouldn't the agent that has something running have a higher score than the one that has nothing running?)

Reply:
> (Fixing this is beyond the scope of this PR).

What do you mean? The tests shouldn't be in the PR?

Reply:
Fixing this problem: that our affinity score is giving an agent with 0/9 the same score as an agent with 1/10.

You can ignore this.